File: src/main/java/org/motivepick/service/UserService.java
package org.motivepick.service;

import org.motivepick.domain.entity.UserEntity;
import org.motivepick.security.Profile;

public interface UserService {

    UserEntity readCurrentUser();

    UserEntity createUserWithTasksIfNotExists(Profile profile, String language);

    void deleteTemporaryUserWithTasks(String accountId);
}

File: src/main/kotlin/org/motivepick/service/UserService.kt
package org.motivepick.service

import org.motivepick.domain.view.UserView
import org.motivepick.security.OAuth2Profile

interface UserService {

    fun readCurrentUser(): UserView?

    fun createUserWithTasksIfNotExists(temporaryAccountId: String, oAuth2Profile: OAuth2Profile, language: String): UserView

    fun deleteTemporaryUserWithTasks(accountId: String)
}
